components/src/dynamo/vllm/tests/test_vllm_worker_handler.py

```
@@ -1854,3 +1854,221 @@ async def boom(_request_id):
+        assert out == pytest.approx([0.1, 0.2, 0.3])
+
+    def test_2d_tensor_keeps_rows(self):
+        out = mod._pooling_output_to_nested(torch.tensor([[1.0, 2.0], [3.0, 4.0]]))
+        assert out == [[1.0, 2.0], [3.0, 4.0]]
+
+    def test_nested_list_passthrough(self):
+        out = mod._pooling_output_to_nested([[1, 2], [3, 4]])
+        assert out == [[1.0, 2.0], [3.0, 4.0]]
+
+    def test_unsupported_type_rejected(self):
+        with pytest.raises(TypeError, match="Unsupported PoolingOutput.data"):
```

Ruff (0.15.21)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)